src/System.Globalization/tests/CultureInfo/CurrentCulture.cs

```diff
@@ -10,21 +10,40 @@ namespace System.Globalization.Tests
     public class Test
     {
         [Fact]
-        public void TestCurrentCultures()
+        public void TestCurrentCulture()
         {
+            // run all tests in one method to avoid multi-threading issues
             CultureInfo defaultCulture = CultureInfo.CurrentCulture;
+            Assert.NotEqual(CultureInfo.InvariantCulture, defaultCulture);
+
             CultureInfo newCulture = new CultureInfo(defaultCulture.Name.Equals("ja-JP", StringComparison.OrdinalIgnoreCase) ? "ar-SA" : "ja-JP");
-            CultureInfo defaultUICulture = CultureInfo.CurrentUICulture;
-            CultureInfo newUICulture = new CultureInfo(defaultCulture.Name.Equals("ja-JP", StringComparison.OrdinalIgnoreCase) ? "ar-SA" : "ja-JP");
+            CultureInfo.CurrentCulture = newCulture;
+            Assert.Equal(CultureInfo.CurrentCulture, newCulture);
 
+            newCulture = new CultureInfo("de-DE_phoneb");
             CultureInfo.CurrentCulture = newCulture;
             Assert.Equal(CultureInfo.CurrentCulture, newCulture);
+            Assert.Equal("de-DE_phoneb", newCulture.CompareInfo.Name);
+
+            CultureInfo.CurrentCulture = defaultCulture;
+            Assert.Equal(CultureInfo.CurrentCulture, defaultCulture);
+        }
 
+        [Fact]
+        public void TestCurrentUICulture()
+        {
+            // run all tests in one method to avoid multi-threading issues
+            CultureInfo defaultUICulture = CultureInfo.CurrentUICulture;
+            Assert.NotEqual(CultureInfo.InvariantCulture, defaultUICulture);
+
+            CultureInfo newUICulture = new CultureInfo(defaultUICulture.Name.Equals("ja-JP", StringComparison.OrdinalIgnoreCase) ? "ar-SA" : "ja-JP");
             CultureInfo.CurrentUICulture = newUICulture;
             Assert.Equal(CultureInfo.CurrentUICulture, newUICulture);
 
-            CultureInfo.CurrentCulture = defaultCulture;
-            Assert.Equal(CultureInfo.CurrentCulture, defaultCulture);
+            newUICulture = new CultureInfo("de-DE_phoneb");
+            CultureInfo.CurrentUICulture = newUICulture;
+            Assert.Equal(CultureInfo.CurrentUICulture, newUICulture);
+            Assert.Equal("de-DE_phoneb", newUICulture.CompareInfo.Name);
```

Comment:
This is pre-existing, but we should probably be resetting the current culture in a finally block.  That way, if there's a bug that causes one of the asserts to fail, subsequent tests won't be thrown off by the change.